### PR TITLE
Add recursive user manifests

### PR DESCRIPTION
**Pull request by markusheinemann**
see #1313 

# How to Test

1. Create the following files inside of `extra-manifests`:

#### **`kustomization.yaml.tpl`**
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - subfolder
```

#### **`subfolder/kustomization.yaml.tpl`**

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - namespace.yaml
```

#### **`subfolder/namespace.yaml.tpl`**

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: mynamespace
```

2. Create a cluster using `terrafom apply`

3. Check if the resource has been applied: `kubectl get ns`

--> It work's on my machine 🤣

### CODE REVIEW

**Review by mysticaltech — CHANGES_REQUESTED** [review]
@markusheinemann Thanks for this. Looking good. Please if you could tweak the docs for this, especially in the examples section in the readme and other relevant places (if this is mentioned there, like kube.tf), it would be great.

**Comment by markusheinemann** [comment]
> @markusheinemann Thanks for this. Looking good. Please if you could tweak the docs for this, especially in the examples section in the readme and other relevant places (if this is mentioned there, like kube.tf), it would be great.

I have updated the related docs and updated the examples as well. I am happy about your feedback.

**Comment by markusheinemann** [comment]
Actual the provided solution will not work due to https://github.com/hashicorp/terraform/issues/16330

File: kustomization_user.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 locals {
-  user_kustomization_templates = try(fileset("extra-manifests", "*.yaml.tpl"), toset([]))
+  user_kustomization_templates = try(fileset("extra-manifests", "**/*.yaml.tpl"), toset([]))
 }
 
 resource "null_resource" "kustomization_user" {
```
Comment (markusheinemann) [inline kustomization_user.tf:5]: Possible solution to solve the terraform file provisioner is to create the directories before copying

```
  provisioner "remote-exec" {
    inline = [
      "mkdir -p $(dirname /var/user_kustomize/${each.key})"
    ]
  }
```

**Review by mysticaltech — APPROVED** [review]
Awesome @markusheinemann! Please let me know when it's working.

**Comment by mysticaltech** [comment]
@markusheinemann What about now?

**Comment by markusheinemann** [comment]
lgtm. I think the PR is done 💯

**Comment by markusheinemann** [comment]
lgtm. I think the PR can be merged in my eyes. Can you please rerun the workflow?

PS. Sorry for closing the PR due to an oversight

**Pull request merged**